iroh-metrics/src/core.rs

```diff
@@ -0,0 +1,114 @@
+use prometheus_client::{encoding::text::encode, registry::Registry};
+use tokio::sync::RwLock;
+
+#[cfg(feature = "bitswap")]
+use crate::bitswap;
+#[cfg(feature = "gateway")]
+use crate::gateway;
+#[cfg(feature = "p2p")]
+use crate::p2p;
+#[cfg(feature = "resolver")]
+use crate::resolver;
+#[cfg(feature = "store")]
+use crate::store;
+
+lazy_static! {
+    pub(crate) static ref CORE: RwLock<Core> = RwLock::new(Core::default());
```

Comment:
Ok, let me explain this one. To make ergonomics simple I needed a global metrics instance with each of the relevant metrics registered on the prometheus registry just once. While not a favorite to have a singleton in rust, makes things work.

The reason this is actually feasible and shareable across threads is that the underlying structures are just references to the prometheus client counters/metrics which are all based on atomic units and data races don't really matter here. Only exception are the histograms but they are currently behind a lock (and might eventually move to a lock free design).

Reply:
You can avoid the rwlock. If you change the `enabled` to be an `AtomicBool` you always only need an immutable reference to `Core`.

Reply:
LOVELY TY!